format/moc3/moc3.go

```diff
@@ -0,0 +1,1766 @@
+		glueInfoSourcesCounts        int64
+	}
+
+	glueInfo struct {
+		weights         int64
+		positionIndices int64
+	}
+
+	glueKeyforms struct {
+		intensities int64
+	}
+
+	warpDeformersV3_3 struct {
+		isQuadSource int64
+	}
+
+	parameterExtensions struct {
+		runtimeSpace0           int64
+		keysSourcesBeginIndices int64
+		keysSourcesCounts       int64
+	}
+
+	warpDeformersV4_2 struct {
+		keyformColorSourcesBeginIndices int64
+	}
+
+	rotationDeformersV4_2 struct {
+		keyformColorSourcesBeginIndices int64
+	}
+
+	artMeshesV4_2 struct {
+		keyformColorSourcesBeginIndices int64
+	}
+
+	keyformMultiplyColors struct {
+		r int64
+		g int64
+		b int64
+	}
+
+	keyformScreenColors struct {
+		r int64
+		g int64
+		b int64
+	}
+
+	parametersV4_2 struct {
+		parameterTypes                                int64
+		blendShapeParameterBindingSourcesBeginIndices int64
+		blendShapeParameterBindingSourcesCounts       int64
+	}
+
+	blendShapeParameterBindings struct {
+		keysSourcesBeginIndices int64
+		keysSourcesCounts       int64
+		baseKeyIndices          int64
+	}
+
+	blendShapeKeyformBindings struct {
+		parameterBindingSourcesIndices               int64
+		blendShapeConstraintIndexSourcesBeginIndices int64
+		blendShapeConstraintIndexSourcesCounts       int64
+		keyformSourcesBlendShapeIndices              int64
+		keyformSourcesBlendShapeCounts               int64
+	}
+
+	blendShapesWarpDeformers struct {
+		targetIndices                               int64
+		blendShapeKeyformBindingSourcesBeginIndices int64
+		blendShapeKeyformBindingSourcesCounts       int64
+	}
+
+	blendShapesArtMeshes struct {
+		targetIndices                               int64
+		blendShapeKeyformBindingSourcesBeginIndices int64
+		blendShapeKeyformBindingSourcesCounts       int64
+	}
+
+	blendShapeConstraintIndices struct {
+		blendShapeConstraintSourcesIndices int64
+	}
+
+	blendShapeConstraints struct {
+		parameterIndices                             int64
+		blendShapeConstraintValueSourcesBeginIndices int64
+		blendShapeConstraintValueSourcesCounts       int64
+	}
+
+	blendShapeConstraintValues struct {
+		keys    int64
+		weights int64
+	}
+
+	blendShapesParts struct {
+		targetIndices                               int64
+		blendShapeKeyformBindingSourcesBeginIndices int64
+		blendShapeKeyformBindingSourcesCounts       int64
+	}
+
+	blendShapesRotationDeformers struct {
+		targetIndices                               int64
+		blendShapeKeyformBindingSourcesBeginIndices int64
+		blendShapeKeyformBindingSourcesCounts       int64
+	}
+
+	blendShapesGlue struct {
+		targetIndices                               int64
+		blendShapeKeyformBindingSourcesBeginIndices int64
+		blendShapeKeyformBindingSourcesCounts       int64
+	}
+}
+
+func decodeMOC3(d *decode.D) any {
+	d.FieldUTF8("magic", 4, d.StrAssert("MOC3"))
+	version := d.FieldU8("version", moc3VersionNames)
+	isBigEndian := d.FieldBoolFn("is_big_endian", func(d *decode.D) bool { return d.U8() != 0 })
```

Comment:
Maybe want to use sym map to a bool? i usually try to make actual the decoded value and then sym the interpretation of the value. That way a user has access to both (`toactual`/`tosym`, default is sym valie if set, otherwise actual)